mungegithub/mungers/submit-queue.go

```diff
@@ -287,13 +287,13 @@ func calcMergeRate(oldRate float64, last, now time.Time) float64 {
 
 // updates a smoothed rate at which PRs are merging per day.
 // returns 'Now()' and the rate.
-// Should be called once after every merge. Also updates sq.totalMerges.
+// Note that this does not update sq.lastMergeTime-- that should be done with
+// every merge, but this function should not be called on retest-not-required
+// merges.
 func (sq *SubmitQueue) updateMergeRate() {
 	now := sq.clock.Now()
 
 	sq.mergeRate = calcMergeRate(sq.mergeRate, sq.lastMergeTime, now)
-	sq.lastMergeTime = now
```

Comment:
hmmmm, moving this is going to throw off the rate calculation when the next NON-retest-not-required comes through, no?  I was using the merge rate to figure out 'how many real PRs it can do a day' but now if the bot blocks for 4 hours, restarts and merges a retest-not-required and then in 30 minutes minutes it will calculate on the last PR having taken 30 minutes instead of 4:30...